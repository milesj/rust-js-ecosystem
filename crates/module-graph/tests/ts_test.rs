mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod ts {
    use super::*;

    #[test]
    fn export_star() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "ts/export-star.ts"));
    }

    #[test]
    fn export_named() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-named.ts"
        ));
    }

    #[test]
    fn export_default_function() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-func.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-func.ts"
        ));
    }

    #[test]
    fn export_default_class() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-class.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-class.ts"
        ));
    }

    #[test]
    fn export_default_reference() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-ref.ts"
        ));
    }

    #[test]
    fn export_default_interface() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-interface.ts"
        ));
    }

    // This doesn't seem to generate exports,
    // even though oxc is a valid enum type!
    #[test]
    fn export_default_enum() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-enum.ts"
        ));
    }
}
