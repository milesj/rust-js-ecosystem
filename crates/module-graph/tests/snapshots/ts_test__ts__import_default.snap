---
source: crates/module-graph/tests/ts_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"ts/import-def.ts\")"
---
{
    2: Module {
        exports: [
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 0,
                        end: 26,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 27,
                        end: 55,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 56,
                        end: 83,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: ValueType,
                        symbol_id: None,
                        name: "Type",
                    },
                ],
                type_only: true,
            },
        ],
        fragment: None,
        id: 2,
        imports: [],
        package_name: None,
        path: "/root/ts/shared.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export const number = 123;\nexport const string = \"abc\";\nexport type Type = boolean;\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 3,
                exports_default: false,
                import_statements: 0,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
    3: Module {
        exports: [
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 0,
                        end: 31,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: DefaultType,
                        symbol_id: None,
                        name: "Foo",
                    },
                ],
                type_only: true,
            },
        ],
        fragment: None,
        id: 3,
        imports: [],
        package_name: None,
        path: "/root/ts/export-def-interface.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export default interface Foo {}\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 1,
                exports_default: true,
                import_statements: 0,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
    4: Module {
        exports: [],
        fragment: None,
        id: 4,
        imports: [],
        package_name: None,
        path: "/root/ts/export-def-enum.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export default enum Status { A, B, C }\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 0,
                exports_default: false,
                import_statements: 0,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [
            Import {
                kind: AsyncStatic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 0,
                    end: 27,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Default,
                        source_name: None,
                        symbol_id: None,
                        name: "foo",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncStatic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 28,
                    end: 70,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Default,
                        source_name: None,
                        symbol_id: None,
                        name: "bar",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncStatic,
                module_id: 3,
                source_request: "./export-def-interface",
                span: Span {
                    start: 91,
                    end: 143,
                },
                symbols: [
                    ImportedSymbol {
                        kind: DefaultType,
                        source_name: None,
                        symbol_id: None,
                        name: "Interface",
                    },
                ],
                type_only: true,
            },
            Import {
                kind: AsyncStatic,
                module_id: 4,
                source_request: "./export-def-enum",
                span: Span {
                    start: 144,
                    end: 186,
                },
                symbols: [
                    ImportedSymbol {
                        kind: DefaultType,
                        source_name: None,
                        symbol_id: None,
                        name: "Enum",
                    },
                ],
                type_only: true,
            },
        ],
        package_name: None,
        path: "/root/ts/import-def.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "import foo from \"./shared\";\nimport { default as bar } from \"./shared\";\n\n// Importing types\nimport type Interface from \"./export-def-interface\";\nimport type Enum from \"./export-def-enum\";\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 0,
                exports_default: false,
                import_statements: 4,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
}
